src/com/TMS/Lesson6/FinderImpl.java

```diff
@@ -0,0 +1,17 @@
+package com.TMS.Lesson6;
+
+public class FinderImpl implements Finder {
+
+    @Override
+    public void find(Director director, String workerName) {
+        for (Employee employee : director.getCountOfWorkers()) {
+            if (employee instanceof Director) {
+                find((Director) employee, workerName);
```

Comment:
I think it is simple searching, but implementation the same as for deep searching